lib/plugins/config/admin.php

```diff
@@ -169,37 +186,33 @@ function settingNaturalComparison($a, $b)
             }
 
             usort($undefined_settings, 'settingNaturalComparison');
-            $this->printH1('undefined_settings', $this->getLang('_header_undefined'));
-            echo '<fieldset>';
-            echo '<div class="table">';
-            echo '<table class="inline">';
+            $this->printHeadline(kHeadlineLevel, 'undefined_settings', $this->getLang('_header_undefined'));
+            echo '<section>';
+            echo '<dl>';
             foreach ($undefined_settings as $setting) {
                 [$label, $input] = $setting->html($this);
-                echo '<tr>';
-                echo '<td class="label">' . $label . '</td>';
-                echo '<td>' . $input . '</td>';
+                echo '<dd class="label">' . $label . '</dd>';
+                echo '<dd class="value">' . $input . '</dd>';
                 echo '</tr>';
             }
-            echo '</table>';
-            echo '</div>';
-            echo '</fieldset>';
+            echo '</dl>';
+            echo '</section>';
         }
 
         // finish up form
-        echo '<p>';
-        echo '<input type="hidden" name="do"     value="admin" />';
-        echo '<input type="hidden" name="page"   value="config" />';
+        echo '<footer>';
+        echo '<input type="hidden" name="do" value="admin">';
+        echo '<input type="hidden" name="page" value="config">';
 
         if (!$this->configuration->isLocked()) {
-            echo '<input type="hidden" name="save"   value="1" />';
-            echo '<button type="submit" name="submit" accesskey="s">' . $lang['btn_save'] . '</button>';
-            echo '<button type="reset">' . $lang['btn_reset'] . '</button>';
+            echo '<input type="hidden" name="save" value="1">';
+			echo '<p><button type="submit" name="submit" accesskey="s">' . $lang['btn_save'] . '</button>';
+            echo '<button type="reset">' . $lang['btn_reset'] . '</button></p>';
         }
 
-        echo '</p>';
-
+        echo '</footer>';
         echo '</form>';
-        echo '</div>';
+        echo '</article>';
```

Comment:
Since you're refactoring, I would like to have this loop be broken up into calling a bunch of class methods instead of doing it all inline.